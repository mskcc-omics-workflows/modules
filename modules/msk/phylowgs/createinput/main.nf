process PHYLOWGS_CREATEINPUT {
    tag "$meta.id"
    label 'process_low'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/phylowgs:v1.5-msk':
        'docker.io/mskcc/phylowgs:v1.5-msk' }"

    input:
    tuple val(meta), path(unfilteredmaf), path(cnv)

    output:
    tuple val(meta), path("cnv_data.txt"), path("ssm_data.txt"), emit: phylowgsinput
    path "versions.yml"                                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    create_phylowgs_inputs.py \\
        --cnvs S1=${cnv} \\
        ${args} \\
        --vcf-type S1=maf S1=${unfilteredmaf}


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch cnv_data.txt
    touch ssm_data.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """
}
