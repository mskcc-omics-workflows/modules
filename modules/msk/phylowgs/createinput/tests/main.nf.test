nextflow_process {

    name "Test Process PHYLOWGS_CREATEINPUT"
    script "../main.nf"
    process "PHYLOWGS_CREATEINPUT"

    tag "modules"
    tag "modules_nfcore"
    tag "phylowgs"
    tag "phylowgs/createinput"

    test("PHYLOWGS_CREATEINPUT - txt") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['unfilteredmaf'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['cnvs_txt'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("PHYLOWGS_CREATEINPUT - txt - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('unfilteredmaf'),
                    file('cnvs_txt')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
