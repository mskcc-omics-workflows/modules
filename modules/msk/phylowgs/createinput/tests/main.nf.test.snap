{
    "PHYLOWGS_CREATEINPUT - txt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ssm_data.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c1620c9a82e06c6ccd1e8465cc928648"
                ],
                "phylowgsinput": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ssm_data.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c1620c9a82e06c6ccd1e8465cc928648"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-03T15:59:43.605713284"
    },
    "PHYLOWGS_CREATEINPUT - txt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,1a3d26484389d5b3b35d20ac5e01f32c",
                        "ssm_data.txt:md5,3a23e8d67f32ea13edb62882952f4122"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c1620c9a82e06c6ccd1e8465cc928648"
                ],
                "phylowgsinput": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cnv_data.txt:md5,1a3d26484389d5b3b35d20ac5e01f32c",
                        "ssm_data.txt:md5,3a23e8d67f32ea13edb62882952f4122"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c1620c9a82e06c6ccd1e8465cc928648"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-03T15:59:38.659729442"
    }
}